acp/identity.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package acp
+
+import (
+	cosmosSecp256k1 "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/sourcenetwork/immutable"
+)
+
+// NoIdentity specifies an anonymous actor.
+var NoIdentity = immutable.None[Identity]()
+
+// Identity uniquely identifies an acp actor.
+type Identity interface {
+	// PublicKey returns the public key of the identity.
+	PublicKey() *secp256k1.PublicKey
+	// Address returns the bech32 address of the identity.
+	Address() string
```

Comment:
question: Why are you returning a `bech32` string instead of a `"github.com/cosmos/cosmos-sdk/crypto/types"`.`Address`?

Reply:
I think it's because `acpSystem.RegisterDocObject` takes a string and that method is used for that.

Reply:
I don't think this matters as address will be replaced with `did` in a follow up work